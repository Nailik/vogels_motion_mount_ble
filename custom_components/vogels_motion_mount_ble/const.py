"""Constants for the Vogels Motion Mount BLE integration."""

DOMAIN = "vogels_motion_mount_ble"

SERVICE_UUID = "3e6fe65d-ed78-11e4-895e-00026fd5c52c"

CHAR_NAME_UUID = "c005fa37-0651-4800-b000-000000000000"
CHAR_SIZE_UUID = "c005fa2b-0651-4800-b000-000000000000"
CHAR_DISTANCE_UUID = "c005fa00-0651-4800-b000-000000000000"
CHAR_ROTATION_UUID = "c005fa01-0651-4800-b000-000000000000"
CHAR_AUTOMATIC_UUID = "c005fa02-0651-4800-b000-000000000000"
CHAR_FREEZE_UUID = "c005fa14-0651-4800-b000-000000000000"
CHAR_PRESET_UUIDS = [
    "c005fa0a-0651-4800-b000-000000000000",
    "c005fa0b-0651-4800-b000-000000000000",
    "c005fa0c-0651-4800-b000-000000000000",
    "c005fa0d-0651-4800-b000-000000000000",
    "c005fa0e-0651-4800-b000-000000000000",
    "c005fa0f-0651-4800-b000-000000000000",
    "c005fa10-0651-4800-b000-000000000000",
]
CHAR_PRESET_UUID = "c005fa2a-0651-4800-b000-000000000000"
