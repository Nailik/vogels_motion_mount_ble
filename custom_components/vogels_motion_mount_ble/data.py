"""Holds the data that is stored in memory by the Vogels Motion Mount Integration."""

from __future__ import annotations

from dataclasses import dataclass
from enum import Enum


class VogelsMotionMountPinSettings(Enum):
    """Pin settings."""

    Deactivated = 12
    Single = 13
    Multi = 15


class VogelsMotionMountAuthenticationType(Enum):
    """Current authentication status."""

    Wrong = 0
    Control = 1
    Full = 2


class VogelsMotionMountAutoMoveType(Enum):
    """Authentication options."""

    Hdmi_1_On = 0
    Hdmi_1_Off = 1
    Hdmi_2_On = 4
    Hdmi_2_Off = 5
    Hdmi_3_On = 8
    Hdmi_3_Off = 9
    Hdmi_4_On = 12
    Hdmi_4_Off = 13
    Hdmi_5_On = 16
    Hdmi_5_Off = 17

    def is_on(self) -> bool:
        """Return True if this enum represents an 'On' state."""
        return self.value % 4 == 0


@dataclass
class VogelsMotionMountAuthenticationStatus:
    """Current authentication status."""

    auth_type: VogelsMotionMountAuthenticationType
    cooldown: int | None = None


@dataclass
class VogelsMotionMountPreset:
    """Preset data."""

    index: int
    data: VogelsMotionMountPresetData | None


@dataclass
class VogelsMotionMountPresetData:
    """Preset data."""

    distance: int
    name: str
    rotation: int


@dataclass
class VogelsMotionMountMultiPinFeatures:
    """Current set of features for authorised user."""

    change_default_position: bool
    change_name: bool
    change_presets: bool
    change_tv_on_off_detection: bool
    disable_channel: bool
    start_calibration: bool


@dataclass
class VogelsMotionMountVersions:
    """Version data."""

    ceb_bl_version: str
    mcp_bl_version: str
    mcp_fw_version: str
    mcp_hw_version: str


@dataclass
class VogelsMotionMountData:
    """Holds the data of the device."""

    automove: VogelsMotionMountAutoMoveType
    connected: bool
    distance: int
    freeze_preset_index: int
    multi_pin_features: VogelsMotionMountMultiPinFeatures
    name: str
    pin_setting: VogelsMotionMountPinSettings
    presets: list[VogelsMotionMountPreset]
    rotation: int
    tv_width: int
    versions: VogelsMotionMountVersions
    permissions: VogelsMotionMountPermissions
    requested_distance: int | None = None
    requested_rotation: int | None = None


@dataclass
class VogelsMotionMountPermissions:
    """Permissions for currently used pin."""

    auth_status: VogelsMotionMountAuthenticationStatus
    change_settings: bool
    change_default_position: bool
    change_name: bool
    change_presets: bool
    change_tv_on_off_detection: bool
    disable_channel: bool
    start_calibration: bool
