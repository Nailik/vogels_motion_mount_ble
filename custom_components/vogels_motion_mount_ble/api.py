"""API Placeholder.

You should create your api seperately and have it hosted on PYPI.  This is included here for the sole purpose
of making this example code executable.
"""

import asyncio
from dataclasses import dataclass, replace
import logging
from .const import CHAR_DISTANCE_UUID, CHAR_ROTATION_UUID, CHAR_PRESET_UUID
from bleak import BleakClient, BleakScanner

_LOGGER = logging.getLogger(__name__)

@dataclass
class VogelsMotionMountData:
    """Class to hold api data."""

    connected: bool = False
    distance: int | None = None
    rotation: int | None = None

class API:
    """Class for example API."""

    def __init__(self, mac: str, pin: str | None, coordinator) -> None:
        """Initialise."""
        self._mac = mac
        self._pin = pin
        self._coordinator = coordinator
        self._data = VogelsMotionMountData(connected=False)
        self._client: BleakClient | None = None
        self._disconnected_event = asyncio.Event()

    async def testConnect(self):
        """Test connection to the BLE device once using BleakClient."""
        try:
            _LOGGER.debug("start test connection")
            async with BleakClient(self._mac, timeout=120) as client:
                _LOGGER.debug("Connected to device: %s", client.is_connected)
        except Exception as err:
            _LOGGER.error("Failed to connect to device: %s", err)
            raise APIConnectionError("Error connecting to api.") from err


    async def perform_action(self):
        _LOGGER.exception("write_gatt_char")
        result = await self._client.write_gatt_char(CHAR_PRESET_UUID, bytes([1]), response=True)
        _LOGGER.exception("write_gatt_char result: %s", result)

    def update(self, **kwargs):
        """Update one or more fields, retaining others from existing data. Then notify the coordinator."""
        self._data = replace(self._data, **kwargs)
        self._coordinator.async_set_updated_data(self._data)

    def _handle_disconnect(self, _):
        _LOGGER.debug("handle Device disconnected!")
        self._disconnected_event.set()

    def _handle_distance_change(self, _, data):
        _LOGGER.debug("distance change %s", data)
        self.update(distance=int.from_bytes(data, "little"))

    def _handle_rotation_change(self, _, data):
        _LOGGER.debug("rotation change %s", data)
        self.update(rotation=int.from_bytes(data, "little"))

    # todo make it optional if the connection should be maintained or connect on command (when required to send command) or poll time
    async def maintainConnection(self):
        """Maintain connection to device."""
        while True:
            _LOGGER.debug("scanning for device %s", self._mac)
            device = await BleakScanner.find_device_by_address(self._mac)

            if device is None:
                _LOGGER.debug("no device %s found, wait then scan again", self._mac)
                await asyncio.sleep(5)
                continue

            try:
                _LOGGER.debug("connecting to device")
                async with BleakClient(device, disconnected_callback=self._handle_disconnect, timeout=120) as client:
                    self.connected = client.is_connected
                    self._client = client
                    _LOGGER.debug("maintain connected to device %s", self._client.is_connected)
                    self.update(connected = self._client.is_connected)

                    self._handle_distance_change(None, await client.read_gatt_char(CHAR_DISTANCE_UUID))
                    self._handle_rotation_change(None, await client.read_gatt_char(CHAR_ROTATION_UUID))

                    await client.start_notify(CHAR_DISTANCE_UUID, self._handle_distance_change)
                    await client.start_notify(CHAR_ROTATION_UUID, self._handle_rotation_change)

                    await self._disconnected_event.wait()
                    # reset event
                    self._disconnected_event.clear()
                    _LOGGER.debug("device disconnected %s", self.connected)
                    self.update(connected = client.is_connected)
                    await asyncio.sleep(1)
            except Exception:
                # catch bleak.exc.BleakError: No backend with an available connection slot that can reach address D9:13:5D:AB:3B:37 was found
                _LOGGER.exception("Exception while connecting/connected")
                await asyncio.sleep(5)


class APIConnectionError(Exception):
    """Exception class for connection error."""
