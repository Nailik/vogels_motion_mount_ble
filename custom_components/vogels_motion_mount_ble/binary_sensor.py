"""Interfaces with the Integration 101 Template api sensors."""

import logging

from homeassistant.components.binary_sensor import (
    BinarySensorDeviceClass,
    BinarySensorEntity,
)
from homeassistant.core import HomeAssistant, callback
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from . import MyConfigEntry
from .base import ExampleBaseEntity
from .const import DOMAIN
from homeassistant.helpers import config_validation as cv
import voluptuous as vol
from homeassistant.helpers.typing import ConfigType
from homeassistant.core import HomeAssistant, ServiceCall, callback

_LOGGER = logging.getLogger(__name__)

async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: MyConfigEntry,
    async_add_entities: AddEntitiesCallback,
):
    """Set up the Binary Sensors."""
    coordinator = hass.data[DOMAIN][config_entry.entry_id]
    async_add_entities([ExampleBinarySensor(coordinator)])


class ExampleBinarySensor(ExampleBaseEntity, BinarySensorEntity):
    """Implementation of a sensor."""

    _attr_device_class = BinarySensorDeviceClass.CONNECTIVITY
    _attr_has_entity_name = True
    _attr_translation_key = "connection"
    _attr_should_poll = True

    @callback
    def _handle_coordinator_update(self) -> None:
        """Update sensor with latest data from coordinator."""
        _LOGGER.exception("_handle_coordinator_update binary sensor")
        # This method is called by your DataUpdateCoordinator when a successful update runs.
        self.async_write_ha_state()

    @property
    def name(self) -> str:
        """Return the name of the sensor."""
        return "connection"

    @property
    def unique_id(self) -> str:
        """Return unique id."""
        # All entities must have a unique id.  Think carefully what you want this to be as
        # changing it later will cause HA to create new entities.
        return f"{DOMAIN}-{self.coordinator.mac}-connected"

    @property
    def is_on(self) -> bool:
        """Return if the binary sensor is on."""
        _LOGGER.exception("is_on binary sensor")
        if not self.coordinator.data:
            _LOGGER.error("No data available in coordinator")
            return None
        # This needs to enumerate to true or false
        return self.coordinator.data.connected

