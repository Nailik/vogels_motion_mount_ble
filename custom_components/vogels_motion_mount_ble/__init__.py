"""Integration for a Vogels Motion Mount via BLE."""

from __future__ import annotations

import logging
from homeassistant.const import __version__ as ha_version
from packaging import version
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import (
    ConfigEntryAuthFailed,
    ConfigEntryError,
    ConfigEntryNotReady,
)

from .api import APIAuthenticationError, APIConnectionDeviceNotFoundError
from .coordinator import VogelsMotionMountBleCoordinator
from .services import async_setup_services

_LOGGER = logging.getLogger(__name__)

PLATFORMS: list[Platform] = [
    Platform.BINARY_SENSOR,
    Platform.BUTTON,
    Platform.NUMBER,
    Platform.SELECT,
    Platform.SENSOR,
    Platform.SWITCH,
    Platform.TEXT,
]

type VogelsMotionMountBleConfigEntry = ConfigEntry[VogelsMotionMountBleCoordinator]

# for this version read_only was added to config selectors
MIN_HA_VERSION = "2025.6.0"

if version.parse(ha_version) < version.parse(MIN_HA_VERSION):
    raise RuntimeError(
        f"Vogels Motion Mount BLE requires Home Assistant {MIN_HA_VERSION}+"
    )


async def async_setup(
    hass: HomeAssistant, entry: VogelsMotionMountBleConfigEntry
) -> bool:
    """Set up Vogels Motion Mount integration services."""
    _LOGGER.debug("async_setup called with config_entry: %s", entry)
    async_setup_services(hass)
    return True


async def async_setup_entry(
    hass: HomeAssistant, config_entry: VogelsMotionMountBleConfigEntry
) -> bool:
    """Set up Vogels Motion Mount Integration from a config entry."""
    _LOGGER.debug("async_setup_entry called with config_entry: %s", config_entry)

    # Registers update listener to update config entry when options are updated.
    unsub_update_listener = config_entry.add_update_listener(async_reload_entry)

    # Initialise the coordinator that manages data updates from your api.
    coordinator = VogelsMotionMountBleCoordinator(
        hass=hass,
        config_entry=config_entry,
        unsub_update_listener=unsub_update_listener,
    )

    try:
        await coordinator.api.refresh_data()
    except APIConnectionDeviceNotFoundError as err:
        raise ConfigEntryNotReady("error_device_not_found") from err
    except APIAuthenticationError as err:
        raise ConfigEntryAuthFailed("error_invalid_athentication") from err
    except Exception as err:
        raise ConfigEntryError(
            translation_key=f"Something went wrong {err}",
            translation_placeholders={"error": err},
        ) from err

    config_entry.runtime_data = coordinator
    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
    return True


async def async_reload_entry(
    hass: HomeAssistant, config_entry: VogelsMotionMountBleConfigEntry
) -> None:
    """Reload config entry."""
    _LOGGER.debug(
        "async_reload_entry async_reload with pin %s", config_entry.data["conf_pin"]
    )
    await hass.config_entries.async_reload(config_entry.entry_id)


async def async_unload_entry(
    hass: HomeAssistant, config_entry: VogelsMotionMountBleConfigEntry
) -> bool:
    """Unload a config entry."""
    _LOGGER.debug("async_unload_entry")
    if unload_ok := await hass.config_entries.async_unload_platforms(
        config_entry, PLATFORMS
    ):
        _LOGGER.debug("async_unload_entry pop")
        coordinator: VogelsMotionMountBleCoordinator = config_entry.runtime_data
        await coordinator.unload()

    return unload_ok
