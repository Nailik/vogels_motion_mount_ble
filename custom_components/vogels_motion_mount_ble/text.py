"""Number entities to define properties that can be changed for Vogels Motion Mount BLE entities."""

from custom_components.vogels_motion_mount_ble.api import (
    SettingsRequestType,
    VogelsMotionMountActionType,
)

from homeassistant.components.text import TextEntity
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from . import VogelsMotionMountBleConfigEntry
from .base import VogelsMotionMountBleBaseEntity, VogelsMotionMountBlePresetBaseEntity
from .coordinator import VogelsMotionMountBleCoordinator
from homeassistant.const import EntityCategory


async def async_setup_entry(
    _: HomeAssistant,
    config_entry: VogelsMotionMountBleConfigEntry,
    async_add_entities: AddEntitiesCallback,
):
    """Set up the TextEntities for name, preset names and pins."""
    coordinator: VogelsMotionMountBleCoordinator = config_entry.runtime_data

    async_add_entities(
        [
            NameText(coordinator),
            *[PresetNameText(coordinator, preset_index) for preset_index in range(7)],
        ]
    )


class NameText(VogelsMotionMountBleBaseEntity, TextEntity):
    """Implementation of a the Name Text."""

    _attr_unique_id = "name"
    _attr_translation_key = _attr_unique_id
    _attr_native_min = 1
    _attr_native_max = 20
    _attr_icon = "mdi:rename-box-outline"
    _attr_entity_category = EntityCategory.CONFIG

    @property
    def native_value(self):
        """Return the state of the entity."""
        return self.coordinator.data.name

    @property
    def available(self) -> bool:
        """Set availability if user has permission."""
        return self.coordinator.api.has_permission(
            action_type=VogelsMotionMountActionType.Settings,
            settings_request_type=SettingsRequestType.change_name,
        )

    async def async_set_value(self, value: str) -> None:
        """Set the name value from the UI."""
        await self.coordinator.api.set_name(value)


class PresetNameText(VogelsMotionMountBlePresetBaseEntity, TextEntity):
    """Implementation of a sensor."""

    _attr_translation_key = "preset_name_custom"
    _attr_native_min = 1
    _attr_native_max = 32
    _attr_icon = "mdi:form-textbox"
    _attr_entity_category = EntityCategory.CONFIG

    def __init__(
        self, coordinator: VogelsMotionMountBleCoordinator, preset_index: int
    ) -> None:
        """Initialize unique_id because it's derived from preset_index."""
        super().__init__(coordinator, preset_index)
        self._attr_unique_id = f"preset_name_{self._prop_preset_index}"

    @property
    def available(self) -> bool:
        """Set availability if preset exists and user has permission."""
        return super().available and self.coordinator.api.has_permission(
            action_type=VogelsMotionMountActionType.Settings,
            settings_request_type=SettingsRequestType.change_presets,
        )

    @property
    def native_value(self):
        """Return the current value."""
        if self._preset:
            return self._preset.name
        return None

    async def async_set_value(self, value: str) -> None:
        """Set the preset name value from the UI."""
        await self.coordinator.api.set_preset(
            preset_index=self._preset_index,
            name=value,
        )
