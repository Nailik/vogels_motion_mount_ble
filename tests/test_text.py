import pytest
from syrupy.assertion import SnapshotAssertion
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er
from unittest.mock import patch

from . import setup_integration

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)
from unittest.mock import AsyncMock
from custom_components.vogels_motion_mount_ble.data import VogelsMotionMountPresetData

from custom_components.vogels_motion_mount_ble.text import (
    NameText,
    PresetNameText,
)

# -------------------------------
# endregion
# region Setup
# -------------------------------


async def test_all_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    with patch("custom_components.vogels_motion_mount_ble.PLATFORMS", [Platform.TEXT]):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


# -------------------------------
# endregion
# region Actions
# -------------------------------


@pytest.mark.asyncio
async def test_name_text_set_value(mock_coord):
    mock_coord.set_name = AsyncMock()
    mock_coord.data.name = "Old Name"

    entity = NameText(mock_coord)

    # native_value reflects coordinator
    assert entity.native_value == "Old Name"

    # Set a new name
    await entity.async_set_value("New Name")
    mock_coord.set_name.assert_awaited_once_with("New Name")


@pytest.mark.asyncio
async def test_preset_name_text_set_value_existing_data(mock_coord):
    preset = mock_coord.data.presets[0]
    preset.data = VogelsMotionMountPresetData(
        name="Preset 1", distance=100, rotation=20
    )
    mock_coord.set_preset = AsyncMock()

    entity = PresetNameText(mock_coord, 0)

    # native_value reflects current preset name
    assert entity.native_value == "Preset 1"

    # Update name → should pass updated preset to coordinator
    await entity.async_set_value("New Preset Name")
    called_preset = mock_coord.set_preset.await_args[0][0]
    assert called_preset.data.name == "New Preset Name"


@pytest.mark.asyncio
async def test_preset_name_text_set_value_no_existing_data(mock_coord):
    preset = mock_coord.data.presets[1]
    preset.data = None
    mock_coord.set_preset = AsyncMock()

    entity = PresetNameText(mock_coord, 1)

    # native_value should be None if no data exists
    assert entity.native_value is None

    # Update name → should create a fresh preset data object
    await entity.async_set_value("Fresh Name")
    called_preset = mock_coord.set_preset.await_args[0][0]
    assert called_preset.data.name == "Fresh Name"
    assert called_preset.data.distance == 0
    assert called_preset.data.rotation == 0


# -------------------------------
# endregion
# -------------------------------
