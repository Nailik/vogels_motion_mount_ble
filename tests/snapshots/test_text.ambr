# serializer version: 1
# name: test_all_entities[text.mount_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 20,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:rename-box-outline',
    'original_name': 'Name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'name',
    'unique_id': 'name',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Name',
      'icon': 'mdi:rename-box-outline',
      'max': 20,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'Mount',
  })
# ---
# name: test_all_entities[text.mount_preset_0_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_0_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 0 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_0',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_0_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 0 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_0_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_1_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_1_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 1 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_1',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_1_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 1 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_1_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_2_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_2_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 2 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_2',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_2_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 2 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_2_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_3_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_3_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 3 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_3',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_3_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 3 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_3_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_4_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_4_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 4 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_4',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_4_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 4 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_4_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_5_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_5_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 5 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_5',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_5_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 5 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_5_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_all_entities[text.mount_preset_6_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'text',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'text.mount_preset_6_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:form-textbox',
    'original_name': 'Preset 6 name',
    'platform': 'vogels_motion_mount_ble',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'preset_name_custom',
    'unique_id': 'preset_name_6',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[text.mount_preset_6_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Mount Preset 6 name',
      'icon': 'mdi:form-textbox',
      'max': 32,
      'min': 1,
      'mode': <TextMode.TEXT: 'text'>,
      'pattern': None,
    }),
    'context': <ANY>,
    'entity_id': 'text.mount_preset_6_name',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
